cmd/cigarToBed/cigarToBed.go

```diff
@@ -0,0 +1,198 @@
+//raven changed the input variable from filename string to inputFile EasyReader, so that the file is only opened 1 time for 2 purposes: faDone and CountSeqIdx
+func CountSeqIdx(inputFile *fileio.EasyReader) int {
+	var line string
+	var seqIdx int = 1 //I know in Read seqIdx is int64 and starts with -1, but I am using it differently here. EasyReader comes in having read the first fasta, so seqIdx starts with 1
+	var doneReading bool = false
+	defer inputFile.Close()
+	for line, doneReading = fileio.EasyNextRealLine(inputFile); !doneReading; line, doneReading = fileio.EasyNextRealLine(inputFile) {
+		if strings.HasPrefix(line, ">") {
+			seqIdx++
+		}
+	}
+	return seqIdx
+}
+
+//raven did not put this helper function into the globalAlignment function because it is used in the globalAlignment function
+//faONe is target, faTwo is query
+func cigarToGraph(target fasta.Fasta, query fasta.Fasta, aln []align.Cigar) *genomeGraph.GenomeGraph {
+	answer := genomeGraph.EmptyGraph()
+	//use targetEnd and queryEnd to track position number for each fasta sequence as we move along.
+	var targetEnd, queryEnd int = 0, 0
+	var curr *genomeGraph.Node
+
+	//Creating the first node. This is done independent of all other number because this node has no 'previous' nodes.  All following code leverages the cigar output (second number printed in the Op position of the struct {}) to determine if the alignment returned a 0:match, 1:insertion, or 2:deletion. All indels are relative to the target sequence.
+	curr, targetEnd, queryEnd = genomeGraph.FaSeqToNode(target, query, targetEnd, queryEnd, aln[0], 0)
+	curr = genomeGraph.AddNode(answer, curr)
+	//Drawing the remaining nodes and all edges. Method for adding edges is based on previous nodes.
+	for i := 1; i < len(aln); i++ {
+		curr, targetEnd, queryEnd = genomeGraph.FaSeqToNode(target, query, targetEnd, queryEnd, aln[i], i)
+		curr = genomeGraph.AddNode(answer, curr)
+		if aln[i].Op == 0 {
+			genomeGraph.AddEdge(&answer.Nodes[i-1], curr, 1)
+			genomeGraph.AddEdge(&answer.Nodes[i-2], curr, 0.5)
+		} else if aln[i].Op == 1 || aln[i].Op == 2 {
+			genomeGraph.AddEdge(&answer.Nodes[i-1], curr, 0.5)
+		} else {
+			log.Fatalf("Error: cigar.Op = %d is unrecognized...\n", aln[i].Op)
+		}
+	}
+	return answer
+}
+
+//raven moved helper functions from main and non-usage functions into this function
+func GlobalAlignment_CigarToBed(inputFileOne *fileio.EasyReader, inputFileTwo *fileio.EasyReader, outFa string, outIns_bed string, outDel_bed string, FirstPos_InsBed int, FirstPos_DelBed int, Chrom string) {
+	//make sure files meet the usage requirements of globalAlignment.go
+	faOne, faDoneOne := fasta.NextFasta(inputFileOne)
+	faTwo, faDoneTwo := fasta.NextFasta(inputFileTwo)
+	fasta.ToUpper(faOne) //raven added these 2 lines to prevent lowercase letters from indexing out of range in align/linearGap.go
+	fasta.ToUpper(faTwo)
+	if faDoneOne || faDoneTwo {
```

Comment:
Just to understanding conceptually.... faDoneOne and faDoneTwo are error handlers, and this OR statement is essentially to check if these variables hold a value. If they do not, there is no error. If fasta.NextFast() didnt work properly we expect  faDoneOne and/or faDoneTwo to have some value, indicating that the input file we failed to read?